single-kernel/lin_reg_coeff.cpp

```diff
@@ -0,0 +1,196 @@
+#include "common.h"
+#include <iostream>
+
+//using namespace cl::sycl;
+namespace s = cl::sycl;
+
+template <typename T> class VecProductKernel;
+template <typename T> class VecReduceKernel;
+
+template <typename T>
+class LinearRegressionCoeffBench
+{
+protected:    
+    std::vector<T> input1;
+    std::vector<T> input2;
+    std::vector<T> output;
+    T coeff_b1;
+    T coeff_b0;
+
+    // Only needed for verification as reduction is done inplace which modifies the input
+    std::vector<T> input1ver;
+    std::vector<T> input2ver;
+    BenchmarkArgs args;
+
+public:
+  LinearRegressionCoeffBench(const BenchmarkArgs &_args) : args(_args) {}
+  
+  void setup() {      
+    // host memory allocation and initialization
+    input1.resize(args.problem_size);
+    input2.resize(args.problem_size);
+    output.resize(args.problem_size, 0);
+
+    input1ver.resize(args.problem_size);
+    input2ver.resize(args.problem_size);
+
+    for (size_t i = 0; i < args.problem_size; i++) {
+       input1ver[i] = input1[i] = 1.0;
+       input2ver[i] = input2[i] = 2.0;
+    }
+  }
+
+  void vec_product(s::buffer<T, 1> &input1_buf, s::buffer<T, 1> &input2_buf, s::buffer<T, 1> &output_buf) {
+    args.device_queue.submit(
+        [&](cl::sycl::handler& cgh) {
+      auto in1 = input1_buf.template get_access<s::access::mode::read>(cgh);
+      auto in2 = input2_buf.template get_access<s::access::mode::read>(cgh);
+ 
+       // Use discard_write here, otherwise the content of the host buffer must first be copied to device
```

Comment:
By the way, when `output_buf` hasn't been written before on the host, a good SYCL runtime will recognize that a data transfer is unnecessary even for a regular `write` access mode since both host and device buffer are in the same state :)